src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -179,9 +181,21 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeIsReadOnlyAttribute(this));
             }
 
-            if (this.IsParams && this.ContainingSymbol is SynthesizedDelegateInvokeMethod)
+            if (this.ContainingSymbol is SynthesizedDelegateInvokeMethod or SynthesizedClosureMethod)
             {
-                AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_ParamArrayAttribute__ctor));
+                if (this.IsParams)
+                {
+                    AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_ParamArrayAttribute__ctor));
+                }
+
+                // Synthesize DecimalConstantAttribute.
+                var defaultValue = this.ExplicitDefaultConstantValue;
+                if (defaultValue != ConstantValue.NotAvailable &&
+                    defaultValue.SpecialType == SpecialType.System_Decimal &&
+                    this.HasDefaultArgumentSyntax)
```

Comment:
>this.HasDefaultArgumentSyntax [](http://example.com/codeflow?start=20&length=29)

What are we trying to accomplish with this check? Is the goal to avoid synthesizing the attribute if there is explicit attribute on the symbol? If so, it would probably make sense to have an API that tells us if we have an explicit DecimalConstant attribute, and its implementation should probably do what SourceParameterSymbolBase.AddSynthesizedAttributes does for the same purpose.

Reply:
> perhaps, SynthesizedComplexParameterSymbol should override AddSynthesizedAttributes and delegate the call to _baseParameterForAttributes if there is one

This wouldn't solve the problem for synthesized delegates as they don't use `_baseParameterForAttributes` (it's `null` for them, it's only used for closure methods).

Reply:
> perhaps, SynthesizedComplexParameterSymbol should override AddSynthesizedAttributes and delegate the call to _baseParameterForAttributes if there is one

This wouldn't solve the problem for synthesized delegates as they don't use `_baseParameterForAttributes` (it's `null` for them, it's only used for closure methods).

Reply:
> This wouldn't solve the problem for synthesized delegates as they don't use _baseParameterForAttributes (it's null for them, it's only used for closure methods).

I wasn't suggesting that they would. The comment was made for specific condition that appeared to be specifically targeting closure methods.

Reply:
Also, the suggestion was specifically for the case when ```_baseParameterForAttributes``` is not null:
> delegate the call to _baseParameterForAttributes if there is one

In other words, the delegation should be conditional.

Reply:
I suggest starting with answering the question:
>What are we trying to accomplish with this check?

Reply:
Thanks, delegated the call to `_baseParameterForAttributes` (and removed `HasDefaultArgumentSyntax` as it's no longer needed).

> Like asserting that the types of parameters are equivalent while taking possible type parameter substitution into account.

Would like to add this, but not sure how. It seems I would like to do `TypeWithAnnotations.Equals(_baseParameterForAttributes.TypeWithAnnotations, TypeCompareKind.ConsiderEverything)` but map type parameters. Not sure how to do this via Roslyn APIs and even if I did, I don't have the type mapping available. Alternatively, I could ignore type parameters (these that are owned by the display class, i.e., `SynthesizedParameterSymbol.ContainingSymbol.ContainingSymbol`, I guess), but not sure how to do the equality check with this customization, either.

Reply:
Nevermind, after running in CI, it turns out that `_baseParameterForAttributes.AddSynthesizedAttributes` behaves differently than `SynthesizedParameterSymbolBase.AddSynthesizedAttributes`, for example:
- the former doesn't check availability of dynamic attribute, crashing `AttributeTests_Dynamic.DynamicLambdaParameterChecksDynamic`,
- nullable attributes are emitted differently, crashing `AttributeTests_Nullable.EmitAttribute_LambdaParameters_01`,
- and probably many more.

So I will revert this change.